llms/mlx_lm/server.py

```diff
@@ -457,8 +510,12 @@ def handle_text_completions(self) -> mx.array:
 
         assert "prompt" in self.body, "Request did not contain a prompt"
         prompt_text = self.body["prompt"]
-
-        prompt = self.tokenizer.encode(prompt_text)
+        logging.debug(f"Prompt type: '{type(prompt_text)}'")
```

Comment:
Support for prompt submitted as a token sequence  instead of a string (would only work if lm-evaluation-harness is invoked with information about the tokenizer)

Reply:
Can you explain this? It looks like it its a list we assume the tokens are already encoded and in the first item of the list? What's in the rest of the list? It seems a bit brittle to me.

Reply:
@awni For the completion API, the prompt can be provided as raw tokens [as well](https://github.com/openai/openai-python/blob/main/src/openai/types/completion_create_params.py#L24)  and the server was not handling that situation properly

Reply:
>  The prompt(s) to generate completions for, encoded as a string, array of strings, array of tokens, or array of token arrays.

Right I get that. It assumes if its a list its tokens and it only takes the first element?

Reply:
So, for the completion API, the inputs are:

> prompt(s) to generate completions for, encoded as a string, array of
>     strings, array of tokens, or array of token arrays.

So, ideally **handle_text_completions**,  **handle_completion**,and **generate_response** would have to change to handle these alternative prompt structures instead of just a single string. The use case I was adding this for, lm-evaluation-harness, can send multiple choice question/answer completions batched in a single HTTP request, in which case the prompt is a list of token sequences when it comes in and each should be completed and end up as a _choices_ ouput with multiple dictionaries, rather than just one with an index of 0 in the response.

Reply:
@chimezie I would prefer to add this change in a follow on PR as a separate thing and discuss it further. I'm not sure what the right thing to do is there, but I don't want to block the rest of this PR on figuring that out.